scripts/AbinsModules/LoadVASP.py

```diff
@@ -0,0 +1,328 @@
+
+            # A1 = (  17.7648860000,   0.0000000000,   0.0000000000)
+            # A2 = (   0.0000000000,  18.0379140000,   0.0000000000)
+            # A3 = (   0.0000000000,   0.0000000000,  18.3144580000)
+            unit_cell = np.empty((3, 3), dtype=float)
+            _ = parser.find_first(file_obj=fd, msg='Lattice vectors:')
+            _ = fd.readline() # skip a line
+
+            for i in range(3):
+                lattice_line = fd.readline().decode("utf-8")
+                float_re = r'\s+(\d+\.\d+)'
+                lattice_re = (r'^\sA' + str(i + 1) + r' = \('
+                              + f'{float_re},{float_re},{float_re}\\)$')
+                match = re.match(lattice_re, lattice_line)
+                if not match:
+                    print(lattice_re, lattice_line)
+                    raise ValueError("Something went wrong while reading lattice vectors from OUTCAR")
+                unit_cell[i, :] = list(map(float, match.groups()))
+            file_data['unit_cell'] = unit_cell
+
+            # getting element symbols from an OUTCAR is a bit cumbersome as
+            # VASP tracks this in groups rather than per-atom
+            ion_counts_line = parser.find_first(file_obj=fd, msg='ions per type =')
+            ion_counts = list(map(int, ion_counts_line.split()[4:]))
+
+            # Get ionic masses; these are reported for each PP, then grouped
+            # together on a line e.g.
+            #    POMASS =  12.01  1.00
+            parser.find_last(file_obj=fd, msg="POMASS")
+            masses_line = fd.readline()
+            ion_count_masses = list(map(float, masses_line.decode("utf-8").split()[2:]))
+            masses = list(chain(*[[mass] * ion_count
+                                  for mass, ion_count in zip(ion_count_masses, ion_counts)]))
+
+        # Re-open file as we need to backtrack to the pseudopotential info.
+        # The rest of the file data can now be gathered in one pass.
+        with open(filename, 'rb') as fd:
+            symbol_lines = [parser.find_first(file_obj=fd, msg='VRHFIN') for _ in ion_counts]
+
+            def _ion_count_or_error(line: bytes) -> str:
+                match = re.match(r'^\s+VRHFIN =(\w+):', line.decode("utf-8"))
+                if match:
+                    return match.groups()[0]
+                else:
+                    raise ValueError("Could not identify VRHFIN value")
+
+            # Get sequence of symbols corresponding to ion counts e.g. C, H
+            ion_count_symbols = map(_ion_count_or_error, symbol_lines)
+            # Expand to full symbol list e.g. [C, C, H, H, H, H]
+            symbols = list(chain(*[[symbol] * ion_count
+                                   for symbol, ion_count in zip(ion_count_symbols, ion_counts)]))
+
+            eig_header = 'Eigenvectors and eigenvalues of the dynamical matrix'
+            parser.find_first(file_obj=fd, msg=eig_header)
+
+            eig_pattern = (r'^\s*\d+\s+f(/i|\s+)=\s+\d+\.\d+\s+THz'
```

Comment:
The reading of the OUTCAR file in general seems a bit brittle as it depends on a lot of string parsing and regex expressions. However given that we are recommending and only documenting the xml format to the users this is probably ok.